lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,25 +120,77 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deeplinkingFields, setDeeplinkingFields] = useState(
+    /** @type {Record<string,string>|null} */ (null)
+  );
+
+  const submit = useCallback(
+    /** @param {Content} content */
+    async content => {
+      async function getDeeplinkingFields() {
+        const data = {
+          ...deeplinkingAPI.data,
+          content,
+          extra_params: {
+            groupSet: groupConfig.useGroupSet ? groupConfig.groupSet : null,
+          },
+        };
+        setDeeplinkingFields(
+          await apiCall({
+            authToken: authToken,
+            path: deeplinkingAPI.path,
+            data,
+          })
+        );
+      }
+      // Set shouldSubmit to true early to get the spinner before any potential apiCall
+      setShouldSubmit(true);
+
+      // When deeplinkingAPI is present we want to call the backend to return the form
+      // fields we'll forward to the LMS to complete the DeepLinking request
+      if (content && deeplinkingAPI && !deeplinkingFields) {
+        try {
+          await getDeeplinkingFields();
+        } catch (error) {
+          setErrorInfo({
+            message: 'Creating deeplinking form fields',
```

Comment:
The message for an error should be a brief description of what failed in the past tense, and something that is short but meaningful to users. For example: "Unable to configure assignment". The details of the error should be included elsewhere in the error modal since you're passing error data.